core/src/main/java/feast/core/service/JobCoordinatorService.java

```diff
@@ -223,6 +223,10 @@ private boolean jobRequiresUpgrade(Job job, Set<Store> stores) {
       return true;
     }
 
+    if (stores.stream().anyMatch(s -> s.getLastUpdated().after(job.getCreated()))) {
```

Comment:
This is a hacky solution of the case when store's subscription was changed but it didn't affect configuration (added or removed Store from the Set, see previous line).
Requires better solution in future.